src/util/sock.cpp

```diff
@@ -105,6 +105,17 @@ int Sock::GetSockOpt(int level, int opt_name, void* opt_val, socklen_t* opt_len)
     return getsockopt(m_socket, level, opt_name, static_cast<char*>(opt_val), opt_len);
 }
 
+int Sock::SetSockOpt(int level, int opt_name, const void* opt_val, socklen_t opt_len) const
+{
+    return setsockopt(m_socket, level, opt_name, static_cast<const char*>(opt_val), opt_len);
+}
+
+bool Sock::SetNoDelay() const
```

Comment:
It's a little bit ugly imo to have both a general `SetSockOpt` and a specific `SetNoDelay` on the same interface. I think we should decide either way:
- Add a general `SetSockOpt` . Move the `SetSockOpt(IPPROTO_TCP, TCP_NODELAY, &on, sizeof(on))` into the caller, or, make .
- Add specific calls for SetNoDelay, ReuseAddr, V6only, etc etc.

Reply:
Good point! There is already a general `SetSockOpt()`, so I used it and ditched the `SetNoDelay()` method.